x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/enrollment/RestCreateEnrollmentTokenAction.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.rest.action.enrollment;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenAction;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenRequest;
+import org.elasticsearch.xpack.core.security.action.enrollment.CreateEnrollmentTokenResponse;
+import org.elasticsearch.xpack.security.rest.action.SecurityBaseRestHandler;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+import static org.elasticsearch.rest.RestRequest.Method.PUT;
+
+/**
+ * Rest endpoint to create an enrollment token
+ */
+public class RestCreateEnrollmentTokenAction extends SecurityBaseRestHandler {
+    private final Settings settings;
+
+    /**
+     * @param settings the node's settings
+     * @param licenseState the license state that will be used to determine if
+     * security is licensed
+     */
+    public RestCreateEnrollmentTokenAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+        this.settings = settings;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(
+            new Route(POST, "/_security/enrollment_token"),
+            new Route(PUT, "/_security/enrollment_token"));
+    }
+
+    @Override
+    public String getName() {
+        return "cluster_enrolment_token_action";
+    }
+
+    @Override
+    protected RestChannelConsumer innerPrepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        if (Node.ENROLLMENT_ENABLED.get(settings) != true) {
+            throw new IllegalStateException("Enrollment mode is not enabled.");
```

Comment:
I think status code `4xx` is more suitable in this scenario than `5xx`. As I understand it, we should use `4xx` to signal client error which can be rectified by the cilent (either change request parameters or in this case enable enrollment first). I think `400` is suitable for this case even though it is generic. Because this failure is not really about permission (`403`) or resource being unavailable (`404`).

Reply:
Thanks @ywangd , `400` looks good to me too.